rbi/stdlib/date.rbi

```diff
@@ -368,7 +368,8 @@ class Date
   # DateTime.new(2001,2,3) - DateTime.new(2001,2,2,12)
   #                          #=> (1/2)
   # ```
-  sig {params(arg0: T.untyped).returns(Date)}
+  sig {params(arg0: Date).returns(Rational)}
+  sig {params(arg0: Numeric).returns(Date)}
```

Comment:
I can't remember which sig is picked as the default in the case that none match, should these two be switched?

Reply:
I don't know that either of them makes sense as the "more default", but I'd be happy to switch them whichever way.

Reply:
My thought was that the previous sig always returned a `Date`, and that it might be better to prefer that version.